### PR TITLE
fix: add item and getByKey

### LINKED ISSUE #263: addItem returns resolved promise object instead of added item with the autoIncrement key
Greetings. I'm using ngx-indexed-db with a ng 11 project with the following packages:

```
"dependencies": {
    "@amcharts/amcharts4": "^4.10.17",
    "@angular/animations": "11.1.1",
    "@angular/cdk": "11.1.1",
    "@angular/common": "11.1.1",
    "@angular/compiler": "11.1.1",
    "@angular/core": "11.1.1",
    "@angular/forms": "11.1.1",
    "@angular/platform-browser": "11.1.1",
    "@angular/platform-browser-dynamic": "11.1.1",
    "@angular/router": "11.1.1",
    "@angular/service-worker": "11.1.1",
    "@fortawesome/fontawesome-free": "5.15.1",
    "@microsoft/signalr": "^3.1.6",
    "@ngx-translate/core": "13.0.0",
    "@ngx-translate/http-loader": "6.0.0",
    "@syncfusion/ej2-angular-buttons": "19.1.63",
    "@syncfusion/ej2-angular-calendars": "19.1.64",
    "@syncfusion/ej2-angular-dropdowns": "19.1.64",
    "@syncfusion/ej2-angular-filemanager": "19.1.63",
    "@syncfusion/ej2-angular-grids": "19.1.64",
    "@syncfusion/ej2-angular-inputs": "19.1.63",
    "@syncfusion/ej2-angular-navigations": "19.1.63",
    "@syncfusion/ej2-angular-popups": "19.1.64",
    "@syncfusion/ej2-layouts": "19.1.63",
    "@syncfusion/ej2-locale": "19.1.54",
    "@syncfusion/ej2-notifications": "19.1.63",
    "angular-archwizard": "^6.1.0",
    "angular-gridster2": "^11.0.0",
    "bootstrap": "4.5.3",
    "classlist": "2.0.0",
    "cldr-data": "^36.0.0",
    "intl": "1.2.5",
    "jwt-decode": "^3.1.1",
    "lodash": "4.17.20",
    "modernizr": "3.11.7",
    "ng-dynamic-component": "8.0.1",
    "ngx-bootstrap": "^6.2.0",
    "ngx-extended-pdf-viewer": "8.3.0",
    "ngx-indexed-db": "^9.1.2",
    "ngx-toastr": "^13.2.1",
    "ngx-webcam": "^0.3.0",
    "print-js": "^1.5.0",
    "rxjs": "~6.6.0",
    "screenfull": "5.0.2",
    "simple-line-icons": "2.5.5",
    "source-sans-pro": "3.6.0",
    "spinkit": "2.0.1",
    "tslib": "^2.0.0",
    "zone.js": "0.11.4"
  },
  "devDependencies": {
    "@angular-devkit/build-angular": "0.1101.1",
    "@angular/cli": "11.1.1",
    "@angular/compiler-cli": "11.1.1",
    "@types/jasmine": "~3.6.0",
    "@types/node": "^12.11.1",
    "codelyzer": "^6.0.0",
    "jasmine-core": "~3.6.0",
    "jasmine-spec-reporter": "~5.0.0",
    "karma": "~6.1.0",
    "karma-chrome-launcher": "~3.1.0",
    "karma-coverage": "~2.0.3",
    "karma-jasmine": "~4.0.0",
    "karma-jasmine-html-reporter": "^1.5.0",
    "protractor": "~7.0.0",
    "ts-node": "~8.3.0",
    "tslint": "~6.1.0",
    "typescript": "~4.1.5"
  }
```

In the context of a store with autoincrement key "Id", with the following setup:

```
      {
        store: veicoliStore,
        storeConfig: { keyPath: 'Id', autoIncrement: true },
        storeSchema: [
          { name: 'VAN', keypath: 'VAN', options: { unique: true } },
          { name: 'Categoria', keypath: 'Categoria', options: { unique: false } },
        ]
      },
```

If I try to do a bulk add (with a single element), everything seems to work fine: the array containing the Id of the single element which was added to the store is correctly returned:

bulkAdd_1
bulkAdd_2

Instead, if I perform a normal insert, with addItem, according to the API specification, I would expect the item with the autoassigned key to be returned. Instead, what I get back looks completely different:

addItem_1
addItem_2

Why is that so? Am I missing something?

Thank you

### PR DESCRIPTION
**Pull request by aparzi**
Fix #263

### CODE REVIEW

**Comment by aparzi** [comment]
Hi @assuncaocharles, 
do you have any news for this PR?

Thanks

**Comment by assuncaocharles** [comment]
Hi @aparzi 

Thanks

Not feeling comfortable adding one more `add` method, maybe we should just refactor the old one

**Comment by aparzi** [comment]
Hi @assuncaocharles,
I can share the thought but I don't know how to refactor the getByKey method. Do you have an idea?

**Comment by assuncaocharles** [comment]
@aparzi 

I meant to refactor the old `add` method

**Comment by aparzi** [comment]
@assuncaocharles 

Ah okay you talk about the first add method (returning the entire added object)? So you only have two: add and bulkAdd right? This could be a great idea.
If I can, I could do it

**Comment by aparzi** [comment]
Hi @assuncaocharles,

I updated the PR leaving only the add method which returns the newly added entity.

File: projects/ngx-indexed-db/src/lib/ngx-indexed-db.service.ts [hunk 1]
```diff
@@ -80,9 +80,13 @@ export class NgxIndexedDBService {
               ? objectStore.add(value, key)
               : objectStore.add(value);
 
-            request.onsuccess = (evt: Event) => {
-              const result = (evt.target as IDBOpenDBRequest).result;
-              resolve((result as unknown) as number);
+            request.onsuccess = async (evt: Event) => {
+              console.log('evt => ', evt);
```
Comment (assuncaocharles) [inline projects/ngx-indexed-db/src/lib/ngx-indexed-db.service.ts]:
```suggestion
```

**Comment by aparzi** [comment]
Hi @assuncaocharles,
remove the console log.

**Comment by aparzi** [comment]
Hi @assuncaocharles,
I updated the PR deleted the console log. Do you have any news for this PR?

Thanks

**Comment by aparzi** [comment]
Hi @assuncaocharles,

any update for this PR?

File: projects/ngx-indexed-db/src/lib/ngx-indexed-db.service.ts [hunk 2]
```diff
@@ -80,9 +80,12 @@ export class NgxIndexedDBService {
               ? objectStore.add(value, key)
               : objectStore.add(value);
 
-            request.onsuccess = (evt: Event) => {
-              const result = (evt.target as IDBOpenDBRequest).result;
-              resolve((result as unknown) as number);
+            request.onsuccess = async (evt: Event) => {
+              const result: any = (evt.target as IDBOpenDBRequest).result;
+              const getRequest: IDBRequest = objectStore.get(result) as IDBRequest<T>;
```
Comment (assuncaocharles) [inline projects/ngx-indexed-db/src/lib/ngx-indexed-db.service.ts:82]: @aparzi  Shouldn't we be using `this.getByKey`?

File: projects/ngx-indexed-db/src/lib/ngx-indexed-db.service.ts [hunk 3]
```diff
@@ -307,9 +281,13 @@ export class NgxIndexedDBService {
...
-              this.getByKey(storeName, key).subscribe((newValue) => {
-                resolve(newValue as T);
-              });
+              const request = objectStore.get(key) as IDBRequest<T>;
+              request.onsuccess = (event: Event) => {
+                resolve((event.target as IDBRequest<T>).result);
+              };
+              request.onerror = (event: Event) => {
+                reject(event);
+              };
```
Comment (assuncaocharles) [inline projects/ngx-indexed-db/src/lib/ngx-indexed-db.service.ts]: Why this change?

File: projects/ngx-indexed-db/src/lib/ngx-indexed-db.service.ts (hunk 2 above)
Comment (aparzi) [reply projects/ngx-indexed-db/src/lib/ngx-indexed-db.service.ts:82]: Hi @assuncaocharles,
The getByKey method if called internally to another method returns promise resolved.

File: projects/ngx-indexed-db/src/lib/ngx-indexed-db.service.ts (hunk 3 above)
Comment (aparzi) [reply projects/ngx-indexed-db/src/lib/ngx-indexed-db.service.ts]: Hi @assuncaocharles,
because this method also had the same bug reported by the issue #263. The getByKey method if called internally to another method returns promise resolved.

**Comment by assuncaocharles** [comment]
@aparzi gotcha, you will have to rebase and then we can merge it

**Comment by aparzi** [comment]
@assuncaocharles update this PR by resolving conflicts.
